lib/ipfs/Ipfs.ts

```diff
@@ -0,0 +1,163 @@
+import * as crypto from 'crypto';
+import * as HttpStatus from 'http-status';
+import * as url from 'url';
+import base64url from 'base64url';
+import FetchResult from '../common/models/FetchResult';
+import FetchResultCode from '../common/enums/FetchResultCode';
+import ICas from '../core/interfaces/ICas';
+import IpfsErrorCode from '../ipfs/IpfsErrorCode';
+import nodeFetch from 'node-fetch';
+import ReadableStream from '../common/ReadableStream';
+import SharedErrorCode from '../common/SharedErrorCode';
+import SidetreeError from '../common/SidetreeError';
+import Timeout from './Util/Timeout';
+
+const multihashes = require('multihashes');
+
+/**
+ * Class that implements the `ICas` interface by communicating with IPFS.
+ */
+export default class Ipfs implements ICas {
+  private fetch = nodeFetch;
+
+  public constructor (public uri: string, private fetchTimeoutInSeconds: number) { }
+
+  public async write (content: Buffer): Promise<string> {
+    // A string that is cryptographically impossible to repeat as the boundary string.
+    const multipartBoundaryString = crypto.randomBytes(32).toString('hex');
```

Comment:
I don't understand the boundary string concept. Any pointers?

Reply:
Neither did I, was forced to implement it as the API demands it.

Official spec: https://tools.ietf.org/html/rfc7578#section-4.1